backend/config/accounts.go

```diff
@@ -22,10 +22,10 @@ import (
 
 // Account holds information related to an account.
 type Account struct {
-	CoinCode      coin.Code              `json:"coinCode"`
-	Name          string                 `json:"name"`
-	Code          string                 `json:"code"`
-	Configuration *signing.Configuration `json:"configuration"`
+	CoinCode       coin.Code              `json:"coinCode"`
+	Name           string                 `json:"name"`
+	Code           string                 `json:"code"`
+	Configurations signing.Configurations `json:"configurations"`
```

Comment:
Did you consider using a map? Might make it easier with detecting existing config hashes:

```go
type Account struct {
  ...
  // SigningConfig is keyed by the configuration hash.
  SigningConfig map[string]signing.Configuration
}

// elsewhere
for _, conf := range configurations {
  if _, exist := account.SigningConfig[conf.Hash()]; exist {
    return ErrAccountAlreadyExists
  }
}
```

Reply:
I did but I don't see a benefit yet. Will do it it appears.

Reply:
Well, to me one immediate benefit is no nested loop. And in general, if there's a lookup-by-signing-config algorithm anywhere, it's just easier with a map instead of constantly looping over configurations slice.

Reply:
and of course it's nicer to read compared to:

![image](https://user-images.githubusercontent.com/25405/115541732-c5fb1b80-a29f-11eb-8a31-259e3d9c5043.png)